### PR TITLE
added a note about platform regexes working in  reverse order

### PR DESCRIPTION
**Pull request by wxtim**
This is a small change with no associated issue

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Does not need tests (Doc change).
- [x] No change log entry required (Internal doc change)
- [x] No documentation update required.

### CODE REVIEW

File: cylc/flow/cfgspec/globalcfg.py [hunk 1]
```diff
@@ -842,13 +842,19 @@
             ``desktop0000, .., desktop9999`` one would define platforms with
             names ``[[bigmachine[12]]]`` and ``[[desktop[0-9]{4}]]``.
 
+            Cylc searches through the list of platform regexes in the reverse
+            of the definition order to allow user defined platforms
+            to override site defined platforms. This means, for example, that
+            if were set ``[[a.*]]`` near the bottom of a configuration any
```
Comment (oliver-sanders) [inline cylc/flow/cfgspec/globalcfg.py:848]: that if were set???

Comment (wxtim) [reply cylc/flow/cfgspec/globalcfg.py:848]:
```suggestion
             if ``[[a.*]]`` were set near the bottom of a configuration any 
```

File: cylc/flow/cfgspec/globalcfg.py [hunk 2]
```diff
@@ -842,13 +842,19 @@
             ``desktop0000, .., desktop9999`` one would define platforms with
             names ``[[bigmachine[12]]]`` and ``[[desktop[0-9]{4}]]``.
 
+            Cylc searches through the list of platform regexes in the reverse
+            of the definition order to allow user defined platforms
+            to override site defined platforms. This means, for example, that 
+            if ``[[a.*]]`` were set near the bottom of a configuration any 
+            platform name beginning "a" would retrn that platform.
```
Comment (hjoliver) [inline cylc/flow/cfgspec/globalcfg.py:849]:
```suggestion
            platform name beginning with "a" would return that platform.
```

**Review by hjoliver — APPROVED** [review]

File: cylc/flow/cfgspec/globalcfg.py [hunk 3]
```diff
@@ -842,13 +842,19 @@
             ``desktop0000, .., desktop9999`` one would define platforms with
             names ``[[bigmachine[12]]]`` and ``[[desktop[0-9]{4}]]``.
 
+            Cylc searches through the list of platform regexes in the reverse
+            of the definition order to allow user defined platforms
+            to override site defined platforms. This means, for example, that
+            if ``[[a.*]]`` were set near the bottom of a configuration any
+            platform name beginning with "a" would return that platform.
```
Comment (oliver-sanders) [inline cylc/flow/cfgspec/globalcfg.py:849]:
```suggestion
            Cylc searches for a matching platform in the reverse
            of the definition order to allow user defined platforms
            to override site defined platforms. This means, for example, that
            if ``[[a.*]]`` were set at the bottom of a configuration any
            platform name beginning with "a" would return that platform.
```

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**